src/components/Blogcard/BlogCard.js

```diff
@@ -0,0 +1,33 @@
+import Image from "next/image";
+import Link from "next/link";
+import React from "react";
+
+export const BlogCard = ({ blogs }) => {
+  const { title, slugs, thumbnail, blogDates, topic } = blogs.fields;
```

Comment:
Maybe you can avoid the destructuring step by providing the component with the exact props needed. That is, transforming the data you get from Contentful _before_ passing it to the component 😉